helix-term/tests/test/commands.rs

```diff
@@ -640,3 +640,28 @@ async fn test_join_selections_space() -> anyhow::Result<()> {
 
     Ok(())
 }
+
+#[tokio::test(flavor = "multi_thread")]
+async fn test_read_file() -> anyhow::Result<()> {
+    let mut file = tempfile::NamedTempFile::new()?;
+    let contents_to_read = String::from("some contents");
+    let output_file = helpers::temp_file_with_contents(&contents_to_read)?;
+    let command = format!(":r {:?}<ret><esc>:w<ret>", output_file.path());
+
+    test_key_sequence(
+        &mut helpers::AppBuilder::new()
+            .with_file(file.path(), None)
+            .build()?,
+        Some(&command),
```

Comment:
The command can be in line here:

```suggestion
        Some(&format!(":r {:?}<ret><esc>:w<ret>", output_file.path())),
```

I think it reads a little easier to have the command in the `test_key_sequence` helper